Python/chapter01/1.6 - String Compression/lomeli_1.6.py

```diff
@@ -0,0 +1,29 @@
+# Implement a method to perform basic string compression using the counts
+# of repeated characters.  If the "compressed" string would not become smaller
+# than the original string, your method should return the original stringself.
+# You can assume that the string has only uppercase and lowercase letters.
+
+def stringCompress(st):
+    result = []
+    cur = st[0]
+    count = 0
+    for i in range(len(st)):
+        if cur == st[i]:
```

Comment:
you can use the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html):

```python
if cur == c:
  count += 1
  continue
# rest of the code
result.append(cur)
...
```